scipy/stats/_continuous_distns.py

```diff
@@ -6241,20 +6242,85 @@ def _entropy(self, c):
     def fit(self, data, *args, **kwds):
         parameters = _check_fit_input_parameters(self, data, args, kwds)
         data, fshape, floc, fscale = parameters
-        if floc is None:
-            return super().fit(data, **kwds)
-        if np.any(data - floc < (fscale if fscale else 0)):
+
+        # ensure that any fixed parameters don't violate constraints of the
+        # distribution before continuing.
+        if floc is not None and np.min(data) - floc < (fscale or 0):
             raise FitDataError("pareto", lower=1, upper=np.inf)
-        data = data - floc
 
+        ndata = data.shape[0]
+
+        def get_shape(scale, location):
+            # The first-order necessary condition on `shape` can be solved in
+            # closed form
+            return ndata / np.sum(np.log((data - location) / scale))
+
+        if floc is fscale is None:
+            # The support of the distribution is `(x - loc)/scale > 0`.
+            # The method of Lagrange multipliers turns this constraint
+            # into an equation that can be solved numerically.
+            # See gh-12545 for details.
+
+            def dL_dScale(shape, scale):
+                # The partial derivative of the log-likelihood function w.r.t.
+                # the scale.
+                return ndata * shape / scale
+
+            def dL_dLocation(shape, location):
+                # The partial derivative of the log-likelihood function w.r.t.
+                # the location.
+                return (shape + 1) * np.sum(1 / (data - location))
+
+            def fun_to_solve(scale):
+                # optimize the scale by setting the partial derivatives
+                # w.r.t. to location and scale equal and solving.
+                location = np.min(data) - scale
+                shape = fshape or get_shape(scale, location)
+                return dL_dLocation(shape, location) - dL_dScale(shape, scale)
+
+            def interval_contains_root(lbrack, rbrack):
+                # return true if the signs disagree.
+                return (np.sign(fun_to_solve(lbrack)) !=
+                        np.sign(fun_to_solve(rbrack)))
+
+            # set brackets for `root_scalar` to use when optimizing over the
+            # scale such that a root is likely between them.
+            brack_start = 1
+            lbrack, rbrack = brack_start / 2, brack_start * 2
+            # if a root is not between the brackets, iteratively expand them
+            # until they include a sign change, checking after each bracket is
+            # modified.
+            while (not interval_contains_root(lbrack, rbrack)
+                   and (lbrack > 0 or rbrack < np.inf)):
+                lbrack /= 2
+                rbrack *= 2
```

Comment:
I think we need to be more precise here. I tested this fit method with data generated from all of the continuous distributions, and `levy_l` produces data that has a good (according to negative log likelihood fn), but not as good as the super method, fit. The reason this happens is that when the bounds iterate, they don't actually include a root.


Using the following script, 

```
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Sun Jan 23 16:44:55 2022

@author: swallan
"""
import hypothesis as h
from hypothesis import given, strategies as st
import numpy.testing as t
import scipy.stats as stats
import numpy as np

def _assert_less_or_close_loglike(dist, data, func, **kwds):
    """
    This utility function checks that the log-likelihood (computed by
    func) of the result computed using dist.fit() is less than or equal
    to the result computed using the generic fit method.  Because of
    normal numerical imprecision, the "equality" check is made using
    `np.allclose` with a relative tolerance of 1e-15.
    """
    mle_analytical = dist.fit(data, **kwds)
    # return mle_analytical
    numerical_opt = super(type(dist), dist).fit(data, **kwds)
    ll_mle_analytical = func(mle_analytical, data)
    ll_numerical_opt = func(numerical_opt, data)
    print(f"mle_analytical: {mle_analytical}")
    print(f"numerical_opt: {numerical_opt}")
    print(f"ll_mle_analytical: {ll_mle_analytical}")
    print(f"ll_numerical_opt: {ll_numerical_opt}")
    print(ll_numerical_opt - ll_mle_analytical)
    assert (ll_mle_analytical <= ll_numerical_opt or
            np.allclose(ll_mle_analytical, ll_numerical_opt, rtol=1e-15))


b, location, scale = .002, 0,1
data = stats.levy_l.rvs(loc=location, scale=scale, size=100,
                          random_state=np.random.default_rng(12345))
args = [data, (stats.pareto._fitstart(data), )]

func = stats.pareto._reduce_func(args, {})[1]
_assert_less_or_close_loglike(stats.pareto, data, func)
```

I debugged the fit method and discovered the results of the bounds to see that this while loop terminated with these bounds:
```
IPdb [36]: lbrack, rbrack
(9.094947017729282e-13, 1099511627776)

IPdb [37]: fun_to_solve(lbrack), fun_to_solve(rbrack)
(inf, -8.673617379884035e-19)
```

and `root_scalar` converges to:
```
IPdb [39]: fun_to_solve(res.root)
-724586124394.904
```

I suspect that there may not be a true zero to converge to, but in that case `root_scalar` should at least return a result with a result that is NEAR zero (to me, at least).

The solution appears to be that before iterating `lbrack` or `rbrack` that we need to ensure that a false "zero crossing" is not established due to an infinity. (as suggested).

```suggestion
            while (not interval_contains_root(lbrack, rbrack)
                   and (lbrack > 0 or rbrack < np.inf)):
                if fun_to_solve(lbrack / 2) != np.inf:
                    lbrack /= 2
                if fun_to_solve(rbrack * 2) != np.inf:
                    rbrack *= 2
```

With this implementation the result is better than the super method, and the root is actual zero: 
```
# mle_analytical: (176588453.45690182, -4398046536342.481, 4398046511103.998)
# numerical_opt: (0.10587718011602715, -25240.345197172937, 1.8617891688491186)
# ll_mle_analytical: 1112.2849127647733
# ll_numerical_opt: 1327.9829459807577

# This number is just the number above copied into fun to solve.
IPdb [41]: fun_to_solve(4398046511103.998)
0.0
```

For good measure I also checked to make sure that there is more on the other side of this root:
```
IPdb [42]: fun_to_solve(4398046511103.998*2)
8.673617379884035e-19
```
The right bracket that was previously found was negative, so I think that this is a true crossing of the x axis.

Reply:
Maybe. Please plot `fun_to_solve`.
(Fortunately, this case is totally pathological. Take a look at the data vs the typical shape of `pareto`. If this is the only problematic case you found, that's great.)

Reply:
>  If this is the only problematic case you found, that's great.

Yes, this is the only one. I need to make sure that continuous basic wasn't skipping any distributions, though. 

Here's the whole `fun_to_solve` from 0 to 4398046511103.998 * 5: 
![alldatafromimagexs](https://user-images.githubusercontent.com/44255917/156206717-b3ea0546-4476-4438-89b4-4f7e1d46b129.png)


and then the crossing (last three datapoints)
![x_last_thre](https://user-images.githubusercontent.com/44255917/156206767-b8b0c26d-2487-4c99-b9be-b20592d03ed7.png)

Reply:
There may be a true zero crossing way out here, but if so, it doesn't look like we can tell it apart from numerical noise.